### PR TITLE
upgrade to stable

### PR DESCRIPTION
**Pull request by boris-brtan**
| Q             | A
| ------------- | ---
|Description, reason for the PR| to improve UPGRADE instructions and try the stable release with project that contains external changes 
|New feature| Yes 
|BC breaks| Yes 
|Fixes issues| many
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: src/Shopsys/ShopBundle/Resources/translations/messages.cs.po [hunk 1]
```diff
@@ -133,6 +136,9 @@ msgstr "Název firmy"
 msgid "Company number"
 msgstr "IČ"
 
+msgid "Company users"
+msgstr "Firemní uživatelé"
```
Comment (DavidOstrozlik) [inline src/Shopsys/ShopBundle/Resources/translations/messages.cs.po]:
```suggestion
msgstr "Uživatelé firmy"
```

Comment (DavidOstrozlik) [inline src/Shopsys/ShopBundle/Resources/translations/messages.cs.po]:
```suggestion
msgstr "Uživatelé firmy"
```

File: src/Shopsys/ShopBundle/Resources/translations/messages.cs.po [hunk 2]
```diff
@@ -289,12 +304,18 @@ msgstr "Zboží chci doručit na jinou adresu než fakturační"
...
+msgstr ""
+
 msgid "Image"
 msgstr "Obrázek"
 
 msgid "Invalid login"
 msgstr "Byly zadány neplatné přihlašovací údaje"
 
+msgid "Is company with multiple users"
+msgstr "Firma s vícero uživately"
```
Comment (DavidOstrozlik) [inline src/Shopsys/ShopBundle/Resources/translations/messages.cs.po]:
```suggestion
msgstr "Firma s více uživateli"
```

File: src/Shopsys/ShopBundle/Resources/translations/messages.cs.po [hunk 3]
```diff
@@ -373,6 +394,9 @@ msgstr "Není z čeho vybírat."
 msgid "Number of items"
 msgstr "Počet položek"
 
+msgid "One or more emails are duplicated or already used, e.g.:"
+msgstr "Jeden nebo více e-mailů je duplikováno nebo se již používá, např.:"
```
Comment (DavidOstrozlik) [inline src/Shopsys/ShopBundle/Resources/translations/messages.cs.po]:
```suggestion
msgstr "Jeden nebo více e-mailů je duplikován nebo se již používá, např.:"
```

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php [hunk 4]
```diff
@@ -76,8 +93,9 @@ public function testRegisteredCustomerCartDoesNotGetDeletedIfItContainsRecentlyA
      */
     private function getProductById($productId)
     {
-        /** @var \Shopsys\ShopBundle\Model\Product\ProductFacade $productFacade */
+        /** @var \Shopsys\FrameworkBundle\Model\Product\ProductFacade $productFacade */
```
Comment (DavidOstrozlik) [inline tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php:96]: why did you change the annotation?

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php [hunk 5]
```diff
@@ -86,9 +104,11 @@ private function getProductById($productId)
      */
     private function getCartFacadeForRegisteredCustomer()
     {
-        /** @var \Shopsys\ShopBundle\Model\Customer\CustomerFacade $customerFacade */
+        /** @var \Shopsys\FrameworkBundle\Model\Customer\CustomerFacade $customerFacade */
```
Comment (DavidOstrozlik) [inline tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php]: why did you change the annotation?

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php [hunk 6]
```diff
@@ -158,7 +208,7 @@ private function createCartFacade(CustomerIdentifier $customerIdentifier)
      */
     private function getCartByCustomerIdentifier(CustomerIdentifier $customerIdentifier)
     {
-        $cartFacade = $this->createCartFacade($customerIdentifier);
+        $cartFacade = $this->getContainer()->get(CartFacade::class);
```
Comment (DavidOstrozlik) [inline tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php]: Why did you change this?

File: tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php [hunk 7]
```diff
@@ -84,34 +94,31 @@ public function testOnKernelResponseRecalculateInputPricesWithoutVat(
...
-        /** @var $currency2 \Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency */
+        /** @var \Shopsys\FrameworkBundle\Model\Pricing\Currency\Currency $currency2 */
         $currency2 = $this->getReference(CurrencyDataFixture::CURRENCY_EUR);
 
         $paymentData = $paymentDataFactory->create();
         $paymentData->name = ['cs' => 'name'];
         $paymentData->pricesByCurrencyId = [$currency1->getId() => $inputPriceWithVat, $currency2->getId() => $inputPriceWithVat];
         $paymentData->vat = $vat;
-        /** @var \Shopsys\FrameworkBundle\Model\Payment\Payment $payment */
+        /** @var \Shopsys\ShopBundle\Model\Payment\Payment $payment */
```
Comment (DavidOstrozlik) [inline tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php]: there is no payment in shopbundle

File: tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php [hunk 8]
```diff
@@ -173,18 +180,19 @@ public function testOnKernelResponseRecalculateInputPricesWithVat(
         $availability = new Availability($availabilityData);
         $em->persist($vat);
         $em->persist($availability);
+
         $paymentData = $paymentDataFactory->create();
         $paymentData->name = ['cs' => 'name'];
         $paymentData->pricesByCurrencyId = [$currency1->getId() => $inputPriceWithoutVat, $currency2->getId() => $inputPriceWithoutVat];
         $paymentData->vat = $vat;
-        /* @var $payment \Shopsys\FrameworkBundle\Model\Payment\Payment */
+        /** @var \Shopsys\ShopBundle\Model\Payment\Payment $payment */
```
Comment (DavidOstrozlik) [inline tests/ShopBundle/Functional/Model/Pricing/InputPriceRecalculationSchedulerTest.php]: there is no payment in shopbundle

**Review by DavidOstrozlik — CHANGES_REQUESTED** [review]
Hi, thanks for the PR. I found a few small things so check them out and I have one more question. Why didn't you change the implementation of picture element as it was the last upgrade instruction?

File: src/Shopsys/ShopBundle/Resources/translations/messages.cs.po (hunk 1 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Resources/translations/messages.cs.po]: fixup! translations from previous PRs are fixed

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php (hunk 4 above)
Comment (boris-brtan) [reply tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php:96]: it is on project-base but also there is no ProductFacade on demoshop, so this was mistake in previous PRs

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php (hunk 5 above)
Comment (boris-brtan) [reply tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php]: based on UPGRADE instructions ....but i see it in demoshop so fixup

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php (hunk 6 above)
Comment (boris-brtan) [reply tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php]: based on instuctions but i revert it back

File: migrations-lock.yml [hunk 9]
```diff
@@ -310,3 +310,15 @@
 20190122130312:
     class: Shopsys\FrameworkBundle\Migrations\Version20190122130312
     skip: false
+20190215092226:
```
Comment (vitek-rostislav) [inline migrations-lock.yml]: seems to me that these migrations should not be a part of this commit

File: CHANGELOG.md [hunk 10]
```diff
@@ -70,6 +70,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
     - phing target build-demo is improved
     - translations from previous changes are fixed
     - promocode naming is unified
+    - datafixtures were copied into ShopBundle\Datafixture namespace
+    - services were created in Resources configs
```
Comment (vitek-rostislav) [inline CHANGELOG.md]: what does this mean?

**Review by vitek-rostislav — COMMENTED** [review]
Hi, the new changes look ok :+1: 

I briefly went through the whole PR and I have 2 more notes on that, please check them out :slightly_smiling_face:

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php (hunk 6 above)
Comment (vitek-rostislav) [reply tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php]: what instruction was that?

File: migrations-lock.yml (hunk 9 above)
Comment (boris-brtan) [reply migrations-lock.yml]: seems like after these changes build-demo-dev could get through the migrations, but OK i put it into the first commit

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php (hunk 6 above)
Comment (boris-brtan) [reply tests/ShopBundle/Functional/Model/Cart/CartFacadeTest.php]: money, especially use the changed files based on commits

File: CHANGELOG.md (hunk 10 above)
Comment (boris-brtan) [reply CHANGELOG.md]: datafixture services were registrered in Resources/config/services/commands.yml and Resources/config/services/data_fixtures.yml

**Review by vitek-rostislav — APPROVED** [review]

**Comment by PetrHeinz** [comment]
Beware that the `ProductController::applyDiscountForPriceFilter` now does nothing as the `Money` class is immutable. If in any doubt, check out the soon-to-be-released docs and upgrade notes for the `Money` class in https://github.com/shopsys/shopsys/pull/870

**Review by DavidOstrozlik — APPROVED** [review]

**Pull request merged**